app/modules/intelligence/agents/custom_agents/agent_logic.py

```
@@ -0,0 +1,189 @@
+            if not isinstance(provider_info, dict):
+                raise ValueError(f"Invalid provider_info type: {type(provider_info)}")
+            
+            try:
+                provider = provider_info['preferred_llm']
+                model_type = provider_info['model_type']
+            except KeyError as ke:
+                raise ValueError(f"Missing required LLM configuration field: {ke}")
+
+            if provider == "openai":
+                api_key = self.secret_manager.get_secret("OPENAI_API_KEY_SYSTEM")
+                llm = LLM(model=model_type, api_key=api_key)
```

Ruff (0.8.2)

Undefined name `LLM`

(F821)